TellCore/TellCoreClient.cs

```diff
@@ -232,6 +239,67 @@ public TellstickResult Down(int deviceId)
         }
 
         /// <summary>
+        /// Use this function to iterate over all sensors.
+        /// </summary>
+        /// <returns>Returns a sensor, if present. Call until Result != Success to enumerate sensors.</returns>
```

Comment:
Consider putting this in a class that inherits IEnumerator<SensorResult>, and creating a list-class that inherits IEnumerable<SensorResult>. Then the user of the API does not need knowledge about how TelldusCore is implemented, but can iterate "as usual" over a collection.